cpp/src/arrow/filesystem/filesystem.cc

```diff
@@ -686,6 +689,16 @@ Result<std::shared_ptr<FileSystem>> FileSystemFromUriReal(const Uri& uri,
     }
     return std::make_shared<LocalFileSystem>(options, io_context);
   }
+  if (scheme == "gs") {
```

Comment:
Can we also accept "gcs" for compatibility with https://gcsfs.readthedocs.io/en/latest/#integration ?

Reply:
Seems reasonable to me.

Reply:
done.